File: self/S2_07052025/Ex2_5.m
% Adding 2 image
clear, clc, close all;
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
pkg load image; %load image package in Octave

%Load test image
Ia = imread('prarie.jpg');
Ib = imread('giraffe.jpg');

% Resize I
Ib1 = imresize(Ib, [size(Ia,1), size(Ia,2)]);

Ic = imadd(Ia, Ib1);

%plot image
FS = 15
figure, clf;

subplot(1, 3, 1), imshow(Ia), title('prarie Image', "fontsize", FS);
subplot(1, 3, 2), imshow(Ib), title('giraffe Image', "fontsize", FS);
subplot(1, 3, 3), imshow(Ic), title('prarie giraffe Image', "fontsize", FS);

saveas(gcf, 'prarie_giraffe.jpg');

File: self/S2_07052025/ConvertPython/Ex2_Python/Ex2_5.py
import cv2
import matplotlib.pyplot as plt

img1 = cv2.imread('prarie.jpg')
img2 = cv2.imread('giraffe.jpg')
img2_resized = cv2.resize(img2, (img1.shape[1], img1.shape[0]))
combined_img = cv2.add(img1, img2_resized)
plt.figure(figsize=(12, 4))
plt.subplot(1, 3, 1), plt.imshow(cv2.cvtColor(img1, cv2.COLOR_BGR2RGB)), plt.title('prarie Image', fontsize=15), plt.axis('off')
plt.subplot(1, 3, 2), plt.imshow(cv2.cvtColor(img2, cv2.COLOR_BGR2RGB)), plt.title('giraffe Image', fontsize=15), plt.axis('off')
plt.subplot(1, 3, 3), plt.imshow(cv2.cvtColor(combined_img, cv2.COLOR_BGR2RGB)), plt.title('prarie giraffe Image', fontsize=15), plt.axis('off')
plt.show()
plt.savefig('prarie_giraffe.jpg')